846905 - [FJ5.8 Bug]Unnecessary error messages are reported by the snmpd about bit check
846906 - [FJ6.3 Bug]Unnecessary error messages are reported by the snmpd about bit check

Upstream commits 43c97bbaaf211f4c7f7c704cfcb901dfbf834874 and 8e5cf2a72a46cd483d22e6290a6c8c4c26af6ad2
diff -up net-snmp-5.5/agent/mibgroup/if-mib/data_access/interface.c.c64 net-snmp-5.5/agent/mibgroup/if-mib/data_access/interface.c
--- net-snmp-5.5/agent/mibgroup/if-mib/data_access/interface.c.c64	2012-09-14 10:16:02.039470080 +0200
+++ net-snmp-5.5/agent/mibgroup/if-mib/data_access/interface.c	2012-09-14 10:16:05.120459643 +0200
@@ -521,7 +521,7 @@ netsnmp_access_interface_entry_update_st
                                        &new_vals->stats.ibytes,
                                        &prev_vals->old_stats->ibytes,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCInOctets to 64bits\n"));
 
         if (new_vals->ns_flags & NETSNMP_INTERFACE_FLAGS_CALCULATE_UCAST) {
@@ -529,14 +529,14 @@ netsnmp_access_interface_entry_update_st
                                            &new_vals->stats.iall,
                                            &prev_vals->old_stats->iall,
                                            &need_wrap_check))
-                NETSNMP_LOGONCE((LOG_ERR,
+                DEBUGMSGTL(("access:interface",
                         "Error expanding packet count to 64bits\n"));
         } else {
             if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.iucast,
                                            &new_vals->stats.iucast,
                                            &prev_vals->old_stats->iucast,
                                            &need_wrap_check))
-                NETSNMP_LOGONCE((LOG_ERR,
+                DEBUGMSGTL(("access:interface",
                         "Error expanding ifHCInUcastPkts to 64bits\n"));
         }
 
@@ -544,49 +544,49 @@ netsnmp_access_interface_entry_update_st
                                        &new_vals->stats.iucast,
                                        &prev_vals->old_stats->iucast,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCInUcastPkts to 64bits\n"));
 
         if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.imcast,
                                        &new_vals->stats.imcast,
                                        &prev_vals->old_stats->imcast,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCInMulticastPkts to 64bits\n"));
 
         if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.ibcast,
                                        &new_vals->stats.ibcast,
                                        &prev_vals->old_stats->ibcast,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCInBroadcastPkts to 64bits\n"));
 
         if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.obytes,
                                        &new_vals->stats.obytes,
                                        &prev_vals->old_stats->obytes,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCOutOctets to 64bits\n"));
 
         if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.oucast,
                                        &new_vals->stats.oucast,
                                        &prev_vals->old_stats->oucast,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCOutUcastPkts to 64bits\n"));
 
         if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.omcast,
                                        &new_vals->stats.omcast,
                                        &prev_vals->old_stats->omcast,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCOutMulticastPkts to 64bits\n"));
 
         if (0 != netsnmp_c64_check32_and_update(&prev_vals->stats.obcast,
                                        &new_vals->stats.obcast,
                                        &prev_vals->old_stats->obcast,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:interface",
                     "Error expanding ifHCOutBroadcastPkts to 64bits\n"));
 
     /*
diff -up net-snmp-5.5/agent/mibgroup/ip-mib/data_access/systemstats_common.c.c64 net-snmp-5.5/agent/mibgroup/ip-mib/data_access/systemstats_common.c
--- net-snmp-5.5/agent/mibgroup/ip-mib/data_access/systemstats_common.c.c64	2012-09-14 10:16:02.000000000 +0200
+++ net-snmp-5.5/agent/mibgroup/ip-mib/data_access/systemstats_common.c	2012-09-14 10:19:27.790772748 +0200
@@ -334,7 +334,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInNoRoutes,
                                        &prev_vals->old_stats->HCInNoRoutes,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInNoRoutes to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -342,7 +342,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutNoRoutes,
                                        &prev_vals->old_stats->HCOutNoRoutes,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutNoRoutes to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -350,7 +350,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutDiscards,
                                        &prev_vals->old_stats->HCOutDiscards,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutDiscards to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -358,7 +358,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutFragReqds,
                                        &prev_vals->old_stats->HCOutFragReqds,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutFragReqds to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -366,7 +366,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutFragOKs,
                                        &prev_vals->old_stats->HCOutFragOKs,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutFragOKs to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -374,7 +374,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutFragFails,
                                        &prev_vals->old_stats->HCOutFragFails,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutFragFails to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -382,7 +382,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutFragCreates,
                                        &prev_vals->old_stats->HCOutFragCreates,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutFragCreates to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -390,7 +390,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInReceives,
                                        &prev_vals->old_stats->HCInReceives,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInReceives to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -398,7 +398,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInOctets,
                                        &prev_vals->old_stats->HCInOctets,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInOctets to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -406,7 +406,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInForwDatagrams,
                                        &prev_vals->old_stats->HCInForwDatagrams,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInForwDatagrams to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -414,7 +414,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInDelivers,
                                        &prev_vals->old_stats->HCInDelivers,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInDelivers to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -422,7 +422,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutRequests,
                                        &prev_vals->old_stats->HCOutRequests,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutRequests to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -430,7 +430,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutForwDatagrams,
                                        &prev_vals->old_stats->HCOutForwDatagrams,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutForwDatagrams to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -438,7 +438,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutTransmits,
                                        &prev_vals->old_stats->HCOutTransmits,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutTransmits to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -446,7 +446,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutOctets,
                                        &prev_vals->old_stats->HCOutOctets,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutOctets to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -454,7 +454,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInMcastPkts,
                                        &prev_vals->old_stats->HCInMcastPkts,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInMcastPkts to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -462,7 +462,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInMcastOctets,
                                        &prev_vals->old_stats->HCInMcastOctets,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInMcastOctets to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -470,7 +470,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutMcastPkts,
                                        &prev_vals->old_stats->HCOutMcastPkts,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutMcastPkts to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -478,7 +478,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutMcastOctets,
                                        &prev_vals->old_stats->HCOutMcastOctets,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutMcastOctets to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -486,7 +486,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCInBcastPkts,
                                        &prev_vals->old_stats->HCInBcastPkts,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCInBcastPkts to 64bits in %s\n",
                     prev_vals->tableName));
 
@@ -494,7 +494,7 @@ netsnmp_access_systemstats_entry_update_
                                        &new_vals->stats.HCOutBcastPkts,
                                        &prev_vals->old_stats->HCOutBcastPkts,
                                        &need_wrap_check))
-            NETSNMP_LOGONCE((LOG_ERR,
+            DEBUGMSGTL(("access:systemstats",
                     "Error expanding HCOutBcastPkts to 64bits in %s\n",
                     prev_vals->tableName));
     }
diff -up net-snmp-5.5/snmplib/int64.c.c64 net-snmp-5.5/snmplib/int64.c
--- net-snmp-5.5/snmplib/int64.c.c64	2009-05-06 23:45:20.000000000 +0200
+++ net-snmp-5.5/snmplib/int64.c	2012-09-14 10:16:05.161459490 +0200
@@ -396,7 +396,7 @@ netsnmp_c64_check32_and_update(struct co
     if ((NULL == need_wrap_check) || (0 != *need_wrap_check)) {
         rc = netsnmp_c64_check_for_32bit_wrap(old_prev_val,new_val, 1);
         if (rc < 0) {
-            snmp_log(LOG_ERR,"c64 32 bit check failed\n");
+            DEBUGMSGTL(("c64","32 bit check failed\n"));
             return -1;
         }
     }
@@ -426,7 +426,7 @@ netsnmp_c64_check32_and_update(struct co
          */
         if ((prev_val->low != new_val->low) ||
             (prev_val->high != new_val->high)) {
-            snmp_log(LOG_ERR, "looks like a 64bit wrap, but prev!=new\n");
+            DEBUGMSGTL(("c64", "looks like a 64bit wrap, but prev!=new\n"));
             return -2;
         }
         else if (NULL != need_wrap_check)
